integration-tests/src/tests/client/cold_storage.rs

```diff
@@ -467,3 +471,109 @@ fn test_initial_copy_to_cold_huge_batch() {
+        .stores(vec![hot_store.clone()])
+        .real_epoch_managers(&genesis.config)
+        .nightshade_runtimes(&genesis)
+        .build();
+
+    let height_delta = env.clients[0].config.gc.gc_num_epochs_to_keep * epoch_length * 3;
+
+    let mut last_hash = *env.clients[0].chain.genesis().hash();
+
+    for h in 1..height_delta {
+        let signer = InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+        for i in 0..5 {
+            let tx = SignedTransaction::send_money(
+                h * 10 + i,
+                "test0".parse().unwrap(),
+                "test1".parse().unwrap(),
+                &signer,
+                1,
+                last_hash,
+            );
+            assert_eq!(env.clients[0].process_tx(tx, false, false), ProcessTxResponse::ValidTx);
+        }
+
+        let block = env.clients[0].produce_block(h).unwrap().unwrap();
+        env.process_block(0, block.clone(), Provenance::PRODUCED);
+        last_hash = *block.hash();
+    }
+
+    let keep_going = std::sync::Arc::new(std::sync::atomic::AtomicBool::new(true));
+
+    copy_all_data_to_cold((*store.cold_db().unwrap()).clone(), &hot_store, 1000000, &keep_going)
+        .unwrap();
+
+    update_cold_head(&*store.cold_db().unwrap(), &hot_store, &(height_delta - 1)).unwrap();
+
+    for h in height_delta..height_delta * 2 {
+        let signer = InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+        for i in 0..5 {
+            let tx = SignedTransaction::send_money(
+                h * 10 + i,
+                "test0".parse().unwrap(),
+                "test1".parse().unwrap(),
+                &signer,
+                1,
+                last_hash,
+            );
+            assert_eq!(env.clients[0].process_tx(tx, false, false), ProcessTxResponse::ValidTx);
+        }
+
+        let block = env.clients[0].produce_block(h).unwrap().unwrap();
+        env.process_block(0, block.clone(), Provenance::PRODUCED);
+        last_hash = *block.hash();
+    }
+
+    let start_cold_head =
+        cold_store.get_ser::<Tip>(DBCol::BlockMisc, COLD_HEAD_KEY).unwrap().unwrap().height;
+
+    let mut near_config = NearConfig::new(nearcore::config::Config::default(), genesis.clone(), near_crypto::KeyFile {
+        account_id: AccountId::from_str("example").unwrap(),
+        public_key: PublicKey::from_str("ed25519:6DSjZ8mvsRZDvFqFxo8tCKePG96omXW7eVYVSySmDk8e").unwrap(),
+        secret_key: SecretKey::from_str("ed25519:3D4YudUahN1nawWogh8pAKSj92sUNMdbZGjn7kERKzYoTy8tnFQuwoGUC51DowKqorvkr2pytJSnwuSbsNVfqygr").unwrap(),
+    }, None).unwrap();
+    near_config.client_config = env.clients[0].config.clone();
+    near_config.config.save_trie_changes = Some(true);
+
+    let epoch_manager = EpochManager::new_arc_handle(store.get_hot_store(), &genesis.config);
+    spawn_cold_store_loop(&near_config, &store, epoch_manager).unwrap();
+    std::thread::sleep(std::time::Duration::from_secs(10));
```

Comment:
Is there any way to speed up this test? On it's own it's not too bad but in general should be avoided to not drag on the overall tests execution time e.g. in buildkite on PRs.